### DISCUSSION TITLE
Extension Router Check if SF user is active?

### DISCUSSION THREAD

**Discussion by SyndrixLane** [amazon-connect/amazon-connect-salesforce-scv#139] [Q&A] [answered]
Hello,

I have successfully configured a new aws connect instance using SCV BYOA - I have installed and configured VMX2 and the Extension router.

I was curious if there was a line of code that I could add to the extension router code/process that when it query checks for the user extension to pull ID if it would also check and verify that the user is active?

I have been admin of a scv with the extension router and when I deactivate a user, the extension router can still find and leave them a voicemail/case even when they are deactivated, the only way to prevent this is to remove their extension when you deactivate them.

Thanks,  I was torn between q/a or idea

**Comment by dcarbonetti** [comment, marked as answer]
you could change line 43 of  awsscv_sf_query.py

from
        `sf_query = "SELECT Id, " + sf_sso_object + " FROM User WHERE Extension ='" + sf_extension + "'"`

to
        `sf_query = "SELECT Id, " + sf_sso_object + " FROM User WHERE IsActive = TRUE AND Extension ='" + sf_extension + "'"`

**Reply by SyndrixLane** [reply]
Thank you!
